airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/parquet_parser.py

```diff
@@ -95,7 +95,17 @@ def file_read_mode(self) -> FileReadMode:
         return FileReadMode.READ_BINARY
 
     @staticmethod
-    def _to_output_value(parquet_value: Scalar, parquet_format: ParquetFormat) -> Any:
+    def _to_output_value(parquet_value: Union[Scalar, DictionaryArray], parquet_format: ParquetFormat) -> Any:
+        """
+        Convert an entry in a pyarrow table to a value that can be output by the source.
+        """
+        if isinstance(parquet_value, DictionaryArray):
+            return ParquetParser._dictionary_array_to_python_value(parquet_value)
+        else:
+            return ParquetParser._scalar_to_python_value(parquet_value, parquet_format)
```

Comment:
@girarda what's your stance on testing this? All the cases are still covered under [these tests](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_parquet_parser.py#L173-L191) (the "special case" test is a bit of a tell) - would you update those tests to call the sub-methods and then test _to_output_value with stubs, or is that way overthinking things? (Jumping back in from java where private methods/implementation explicitly is _not_ tested is messing with my brain)

Reply:
I generally agree with not testing private methods. Let's not update update the tests for an implementation detail

Reply:
Cool, that was my take too. Thanks!